### PR TITLE
mgr/telemetry: obscure entity_name with a salt

### PR DESCRIPTION
**Pull request by liewegas**
This is on top of #28847; see final commit

### CODE REVIEW

**Comment by liewegas** [comment]
@l-mb

File: src/pybind/mgr/telemetry/module.py [hunk 1]
```diff
@@ -204,6 +211,13 @@ def gather_crashinfo(self):
                 continue
             c = json.loads(crashinfo)
             del c['utsname_hostname']
+            (etype, eid) = c.get('entity_name', '').split('.')
+            if etype != 'osd':
+                m = hashlib.sha1()
+                m.update(self.salt.encode('utf-8'))
+                m.update(eid.encode('utf-8'))
+                m.update(self.salt.encode('utf-8'))
```
Comment (dmick) [inline src/pybind/mgr/telemetry/module.py]: twice for fun, or typo?

Comment (liewegas) [reply src/pybind/mgr/telemetry/module.py]: i seem to recall this is a standard pattern when doing HMAC to put the salt or hash both before and after as a confounder.  probably doesn't really matter.

Comment (liewegas) [reply src/pybind/mgr/telemetry/module.py]: mostly for fun i guess :)

**Comment by l-mb** [comment]
Looks good to me!

**Pull request merged**